src/middlewared/middlewared/plugins/cloud_backup/snapshot.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime
+import json
+import subprocess
+
+from middlewared.plugins.cloud_backup.restic import get_restic_config
+from middlewared.schema import accepts, Bool, Datetime, Dict, Int, List, returns, Str
+from middlewared.service import CallError, Service
+from middlewared.validators import NotMatch
+
+
+class CloudBackupService(Service):
+
+    class Config:
+        cli_namespace = "task.cloud_backup"
+        namespace = "cloud_backup"
+
+    @accepts(Int("id"))
+    @returns(
+        List("cloud_backup_snapshots", items=[
+            Dict(
+                "cloud_backup_snapshot",
+                Str("id"),
+                Str("hostname"),
+                Datetime("time"),
+                List("paths", items=[Str("path")]),
+                additional_attrs=True,
+            ),
+        ]),
+    )
+    def list_snapshots(self, id_):
```

Comment:
Have we tested under large snapshot counts to see whether this needs to be a job?

Reply:
10000 snapshots take 30 seconds to list, I doubt any reasonable use will go above that.

Reply:
Are these 10k "restic" snapshots or zfs snapshots?

Reply:
Restic snapshots, ZFS snapshots are not involved here